utils/s2n_socket.c

```diff
@@ -204,8 +204,12 @@ int s2n_socket_write(void *io_context, const uint8_t *buf, uint32_t len)
     }
 
     /* On success, the number of bytes written is returned. On failure, -1 is
-     * returned and errno is set appropriately. */
-    return write(wfd, buf, len);
+     * returned and errno is set appropriately. 
+     * Don't generate a SIGPIPE signal if the peer on a stream-
+     * oriented socket has closed the connection.  The EPIPE error is
+     * still returned.
+     */
+    return send(wfd, buf, len, MSG_NOSIGNAL);
```

Comment:
why this change?

applications can chose to ignore `SIGPIPE` themselves.

Reply:
> why this change?
> 
> applications can chose to ignore `SIGPIPE` themselves.

valgrind seems to terminate client process because of SIGPIPE. I can skip EXPECT_EQUAL(status, 0) checking for valgrind test or do something to valgrind though.